rest_tools/client/device_client.py

```diff
@@ -185,22 +185,30 @@ def SavedDeviceGrantAuth(
         retries (int): number of retries to attempt (optional)
     """
     logger = logging.getLogger('SavedDeviceGrantAuth')
-
     filepath = Path(filename)
-    refresh_token = _load_token_from_file(filepath)
-
-    if not refresh_token:
-        auth = OpenIDAuth(token_url)
-        if not auth.provider_info:
-            raise RuntimeError('Token service does not support .well-known discovery')
-        if 'device_authorization_endpoint' not in auth.provider_info:
-            raise RuntimeError('Device grant not supported by server')
-        endpoint = auth.provider_info['device_authorization_endpoint']
-        refresh_token = _perform_device_grant(logger, endpoint, auth.token_url, client_id, client_secret, scopes)
 
     def update_func(access, refresh):
         _save_token_to_file(filepath, refresh)
 
+    refresh_token = _load_token_from_file(filepath)
+    if refresh_token:
+        try:
+            # this will try to refresh, and raise if it fails
+            return OpenIDRestClient(address=address, token_url=token_url, client_id=client_id,
+                                    client_secret=client_secret, refresh_token=refresh_token,
+                                    update_func=update_func, **kwargs)
+        except Exception:
+            pass
```

Comment:
Not really, since it's catching this exception:
https://github.com/WIPACrepo/rest-tools/blob/master/rest_tools/client/openid_client.py#L79

I guess I could change that, if you have preferences?

Reply:
I'd like that more for sure. Could you re-raise the `HTTPError`?

Reply:
I'm hesitant to raise specific error types from dependent libraries, since they tend to have a lot of detail that means nothing to the user.  In this case, the user probably doesn't understand http error codes, but can understand "the token request failed, so something is wrong with your auth."

I guess we could define our own error classes, but I've never been a huge fan of them.

Reply:
If there's a different issue (not an expired token) would you expect this to also happen for the following device code process? I can think of a scenario where the user's connection goes down.

Reply:
If so, then we should be good not bubbling up that kind of error

Reply:
Yes, any sort of connection error I'd expect to also happen in the device code process.  And if it happens to be a strange transient error (a single dropped connection), I'm fine very occasionally prompting for the device code again.

Note that for cron jobs, the device code grant expires after 10 minutes (currently set in Keycloak), so they will eventually time out if you don't login.  If this truly is a transient error, we don't delete the on-disk token, so it can try again next time.